### PR TITLE
riscv_cov_instr class 

### PR DESCRIPTION
**Pull request by hodjat91**
- Completed riscv_cov_instr class (decoupled from riscv_instr_cov_test file)
- Added private _riscv_cov_instr module to manually retrieve format/category/group/imm_t based on the name of the instruction

### CODE REVIEW

File: pygen/pygen_src/isa/_riscv_cov_instr.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
+from pygen.pygen_src.riscv_instr_pkg import (riscv_instr_name_t,
```
Comment (taoliug) [inline pygen/pygen_src/isa/_riscv_cov_instr.py:1]: please add license header for all files

Comment (hodjat91) [reply pygen/pygen_src/isa/_riscv_cov_instr.py:1]: sure, added.

File: pygen/pygen_src/isa/riscv_cov_instr.py [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+        if self.category == riscv_instr_category_t.STORE:
+            if (pre_instr.category == riscv_instr_category_t.STORE and
+                    pre_instr.mem_addr.get_val() == self.mem_addr.get_val()):
+                self.lsu_hazard = hazard_e["WAW_HAZARD"]
+            elif (pre_instr.category == riscv_instr_category_t.LOAD and
+                  pre_instr.mem_addr.get_val() == self.mem_addr.get_val()):
+                self.lsu_hazard = hazard_e["WAR_HAZARD"]
+            else:
+                self.lsu_hazard = hazard_e["NO_HAZARD"]
+        logging.info("Pre: {}, Cur: {}, Hazard: {}/{}".format(
```
Comment (udinator) [inline pygen/pygen_src/isa/riscv_cov_instr.py:306]: I'd suggest making this a `logging.debug()` call instead

Comment (hodjat91) [reply pygen/pygen_src/isa/riscv_cov_instr.py:306]: Sure, thanks Udi.

File: pygen/pygen_src/isa/riscv_cov_instr.py [hunk 3]
```diff
@@ -0,0 +1,530 @@
...
+                self.rs1_value.set_val(self.get_gpr_state(operands[0]))
+                self.rs2 = self.get_gpr(operands[1])
+                self.rs2_value.set_val(self.get_gpr_state(operands[1]))
+                self.imm.set_val(get_val(operands[2]))
+        elif self.format.name == "R_FORMAT":
+            if self.has_rs2 or self.category.name == "CSR":
+                assert len(operands) == 3
+            else:
+                assert len(operands) == 2
+            if self.catefory.name == "CSR":
```
Comment (udinator) [inline pygen/pygen_src/isa/riscv_cov_instr.py:427]: nit: `self.catefory.name` -> `self.category.name`

Comment (hodjat91) [reply pygen/pygen_src/isa/riscv_cov_instr.py:427]: Fixed!

File: pygen/pygen_src/isa/riscv_cov_instr.py (hunk 2 above)
Comment (hodjat91) [reply pygen/pygen_src/isa/riscv_cov_instr.py]: I changed it it to a debug message, it prints prev/current PCs, instruction names, and also type of the hazards. Adding related registers/their values will depend on the hazard type (which would make the code a bit ugly and large), so I didn't include them in the debug message (I think having PC address as well as instruction names/hazard type will be enough, right?)

File: pygen/pygen_src/isa/riscv_cov_instr.py [hunk 4]
```diff
@@ -0,0 +1,508 @@
...
+
+    def get_instr_name(self):
+        get_instr_name = self.instr.name
+        for i in get_instr_name:
+            if i == "_":
+                get_instr_name = get_instr_name.replace(i, ".")
+        return get_instr_name
+
+    def update_src_regs(self, operands):
+        if self.format.name in ["J_FORMAT", "UORMAT"]:
```
Comment (udinator) [inline pygen/pygen_src/isa/riscv_cov_instr.py:362]: typo?

File: pygen/pygen_src/isa/riscv_cov_instr.py (hunk 2 above)
Comment (udinator) [reply pygen/pygen_src/isa/riscv_cov_instr.py]: Yeah, I think that is fine - I think with the provided information it should be straightforward enough to refer to the CSV file and find the two instructions that cause the hazard.

File: pygen/pygen_src/isa/riscv_cov_instr.py (hunk 4 above)
Comment (hodjat91) [reply pygen/pygen_src/isa/riscv_cov_instr.py:362]: Yup, fixed.

**Pull request merged**